runtime/autoload/netrw.vim

```diff
@@ -5570,12 +5469,52 @@ fun! netrw#GX()
   if &ft == "netrw"
    let fname= s:NetrwGetWord()
   else
-   let fname= expand((exists("g:netrw_gx")? g:netrw_gx : '<cfile>'))
+   let fname= exists("g:netrw_gx")? expand(g:netrw_gx) : s:GetURL()
   endif
 "  call Dret("netrw#GX <".fname.">")
   return fname
 endfun
 
+fun! s:GetURL() abort
+  " markdown URL such as [link text](http://ya.ru 'yandex search')
+  try
+    let save_view = winsaveview()
+    if searchpair('\[.\{-}\](', '', ')\zs', 'cbW', '', line('.')) > 0
+      return matchstr(getline('.')[col('.')-1:], '\[.\{-}\](\zs'..s:regex_url..'\ze\(\s\+.\{-}\)\?)')
+    endif
+  finally
+    call winrestview(save_view)
+  endtry
+  " HTML URL such as <a href='http://www.python.org'>Python is here</a>
+  "                  <a href="http://www.python.org"/>
+  try
+    let save_view = winsaveview()
+    if searchpair('<a\s\+href=', '', '\%(</a>\|/>\)\zs', 'cbW', '', line('.')) > 0
+      return matchstr(getline('.')[col('.') - 1 : ],
+            \ 'href=["'.."'"..']\?\zs\S\{-}\ze["'.."'"..']\?/\?>')
+    endif
+    finally
+      call winrestview(save_view)
+    endtry
+  " URL with leading or closing parenthesis (http://google.com)
+  let URL = matchstr(expand("<cWORD>"), '^(\?\zs'..s:regex_url..'\ze)\?$')
+  if !empty(URL) | return URL | endif
+
+  " Is it a file in the current work dir ...
+  let file = expand("<cfile>")
+  if filereadable(file) | return file | endif
+  " ... or in that of the current buffer?
+  let path = expand('%')
+  if isdirectory(path)
+    let dir = fnamemodify(path, ':p')
+  elseif filereadable(path)
+    let dir = fnamemodify(path, ':p:h')
+  endif
+  if exists('dir') && filereadable(dir..'/'..file) | return dir..'/'..file | endif
+
+  return ''
+endf
+
```

Comment:
Yes, a matter of taste. I deemed markdown links common enough, but these can be deferred to a suitable ftplugin

Reply:
Mind you that you can already override the selection to be opened, and this option is preserved

Reply:
> Mind you that you can already override the selection to be opened, and this option is preserved

What do you mean?

Reply:
I meant `:help g:netrw_gx` though I find it rather coarse and in general maybe some options are no longer that useful. I dared to remove `g:netrw_browsex_viewer` as it relied by default on ancient applications such as Netscape and seemed overengineered for its user base (that I suspected to be small since by now somebody would have replaced Netscape Navigator by something more modern).